### PR TITLE
Automated graph validation, starting point

### PR DESCRIPTION
**Pull request by lcpopa**
Signed-off-by: Liviu Popa <liviu-constantin.popa@ing.com>

### CODE REVIEW

**Comment by planetf1** [comment]
The grade build is failing after the version bump. It is not due to this pr. I have now disabled the mandatory nature of the check and will look at resolving. . The best push or update should be fine. Apologies

**Comment by lcpopa** [comment]
recheck

**Comment by planetf1** [comment]
The open lineage janus connector is failing on a rat check
```
[INFO] Rat check: Summary over all files. Unapproved: 1, unknown: 1, generated: 0, approved: 20 licenses.
```

Your second file has no SPDX headers - if you add them in hopefully all will be ok (ignore the gradle issue - my bad)

**Comment by planetf1** [comment]
The spdx header change looks good, but now failing on dependency checks:

```
[WARNING] Used undeclared dependencies found:
[WARNING]    org.glassfish:javax.json:jar:1.0:compile
[INFO] Add the following to your pom to correct the missing dependencies: 
[INFO] 
<dependency>
  <groupId>org.glassfish</groupId>
  <artifactId>javax.json</artifactId>
  <version>1.0</version>
</dependency>
```

It's probably worth making a full build clean maybe even with a cleaned ~/.m2 in case your tests are picking up something bad in the cache?

Also rather than glass fish

javax.json » javax.json-api

is I think better - to avoid pulling in a new group - if you need this new support. 1.1.4 is current - see https://mvnrepository.com/artifact/javax.json/javax.json-api/1.1.4 . No changes for a while. hopefully no CVEs :-)

File: open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/test/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/lineagegraph/OpenLineageGraphValidation.java [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        log.info("Validated " + TABLES.size() + " tables");
+
+        COLUMNS.forEach(this::validateColumn);
+        log.info("Validated " + COLUMNS.size() + " columns");
+    }
+
+    private void validateProcess(Process process){
+        Vertex processAsVertex = g.V().has(VERTEX_QUALIFIED_NAME, process.getQualifiedName()).has(PROPERTY_KEY_LABEL, PROCESS).next();
+
+        assert processAsVertex != null : "Process not found by qualified name " + process;
```
Comment (popa-raluca) [inline open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/test/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/lineagegraph/OpenLineageGraphValidation.java:184]: You might want to print process.getQualifiedName() here

File: open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/test/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/lineagegraph/OpenLineageGraphValidation.java [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+        if( !g.V(columnAsVertex.id()).has(PROPERTY_KEY_LABEL, TABULAR_COLUMN).in(ATTRIBUTE_FOR_SCHEMA)
+                .in("AssetSchemaType").hasLabel(DATA_FILE).hasNext() ){
+            // tabular columns that do not reach a DataFile are not eligible for validation
+            return;
+        }
+
+        boolean isInput = g.V(columnAsVertex.id()).out(COLUMN_DATA_FLOW).hasLabel("subProcess").hasNext();
+        boolean isOutput = g.V(columnAsVertex.id()).in(COLUMN_DATA_FLOW).hasLabel("subProcess").hasNext();
+
+        assert isInput || isOutput : "Column missing input and output";
```
Comment (popa-raluca) [inline open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/open-lineage-connectors/open-lineage-janus-connector/src/test/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/openlineageconnectors/janusconnector/lineagegraph/OpenLineageGraphValidation.java:268]: This check can be split into two different ones, for input/output with specific messages

**Comment by lcpopa** [comment]
Unfortunately, I had to add the glassfish dependency as I couldn't find a workaround. Someone else might try with a better outcome

**Review by popa-raluca — APPROVED** [review]

**Comment by planetf1** [comment]
I took a look and agree.

Glassfish is actually coming in through this too ....

```
[INFO] +- org.janusgraph:janusgraph-core:jar:0.5.2:compile
[INFO] |  +- org.apache.tinkerpop:tinkergraph-gremlin:jar:3.4.8:compile
[INFO] |  +- org.glassfish:javax.json:jar:1.0:test
```

so it's already a dependency in this area of code, so let's stick with it. Thanks for checking

**Review by planetf1 — APPROVED** [review]

**Pull request merged**